### ISSUE TITLE
issue: select in fieldarray with dynamic option list gets confused on item removal

### ISSUE THREAD

**Issue by krausest** [react-hook-form/react-hook-form#9728]
### Version Number

7.4.1

### Codesandbox/Expo snack

https://codesandbox.io/s/clever-morse-55pkuh?file=/src/App.tsx

### Steps to reproduce

1. Go to https://codesandbox.io/s/clever-morse-55pkuh?file=/src/App.tsx (based on V7 TS Template)
2. Preparation: Notice that the second row has 'B' selected, third row has 'C' selected. The list of options is computed dynamically according to the rule that it must not show values that are selected in the other select boxes.
3. Click on delete button next to the select showing 'B'
4. The row with 'B' is removed, but the value of the 'C' row now shows incorrectly has 'D' selected (fun fact: Safari shows 'B'). The state printed next to the select box shows that this is out of sync with the form state and it should still show 'C'.

There's essential the same code with the latest react-hook-form version 7.41.3 and a pure react implementation below it on: https://codesandbox.io/s/loving-jerry-ly185q?file=/src/App.tsx:2114-2118
This version has the same issue.

### Expected behaviour

In step 4 the second line should show 'C'.

When I convert the select to a controlled object (by simply setting its value) it works as expected, but I'd say this shouldn't be necessary.
When the list of options is static (by removing the filter operation in line 46) the selection of the field array shows the correct value. 

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
I don't think this is an issue related to react hook form.  since your input is uncontrolled:

```tsx
<option key={c.value} value={c.value}>
  {c.label}
</option>
 ```

Your key is going to get duplicated, I would consider building a controlled component for your usage instead especially since there is runtime on each row update with your select options.
